src/HTSPTypes.cpp

```diff
@@ -437,3 +441,378 @@ void AutoRecording::SetFulltext(uint32_t fulltext)
 }
 
 } // namespace htsp
+
+/* **************************************************************************
+ * Subscription
+ * *************************************************************************/
+
+Subscription::Subscription(CHTSPConnection &conn) :
+  m_channelId(0),
+  m_speed(0),
+  m_weight(0),
+  m_subscriptionId(0),
+  m_state(SUBSCRIPTION_STOPPED),
+  m_conn(conn)
+{
+}
+
+bool Subscription::IsActive() const
+{
+  return (GetState() != SUBSCRIPTION_STOPPED);
+}
+
+eSubsriptionState Subscription::GetState(void) const
+{
+  /* used in multiple threads */
+  CLockObject lock(m_mutex);
+  return m_state;
+}
+
+uint32_t Subscription::GetId() const
+{
+  /* used in multiple threads */
+  CLockObject lock(m_mutex);
+  return m_subscriptionId;
+}
+
+uint32_t Subscription::GetChannelId() const
+{
+  return m_channelId;
+}
+
+uint32_t Subscription::GetSpeed() const
+{
+  return m_speed;
+}
+
+uint32_t Subscription::GetWeight() const
+{
+  /* used in multiple threads */
+  CLockObject lock(m_mutex);
+  return m_weight;
```

Comment:
@Jalle19 thanks for your help on this.

m_weght, m_state and m_subscriptionId are made threadsafe in the right way I guess? :p